CMakeLists.txt

```diff
@@ -33,6 +33,7 @@ option(BENCHMARK_DOWNLOAD_DEPENDENCIES "Allow the downloading and in-tree buildi
 # in cases where it is not possible to build or find a valid version of gtest.
 option(BENCHMARK_ENABLE_GTEST_TESTS "Enable building the unit tests which depend on gtest" ON)
 
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
Move it higher, and make it a cache variable:
```
set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ${BUILD_SHARED_LIBS} CACHE BOOL "")
```

Reply:
Why a cache variable? This only takes effect when building with `BUILD_SHARED_LIBS` on Windows - otherwise it does nothing. See [WINDOWS_EXPORT_ALL_SYMBOLS](https://cmake.org/cmake/help/latest/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html).

Reply:
I think it may be best to force users hand if the `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` was already specified?

Reply:
This variable is meant to be set by the project, not by the user. It's not intended to be a user-toggleable feature, but merely a convenience to avoid having to prefix everything with `__declspec(dllexport)`.

Reply:
I'm also not seeing the benefit of making this a user-settable cache variable. I don't think you would ever want to build gRPC DLLs that export no symbols, would you?